File: Projects/rosalind/REVC.java
import java.io.*;
import java.util.*;

public class REVC {
	private static final Map<Character, Character> COMPLEMENT = new HashMap<Character, Character>() {{
		put('A', 'T'); put('C', 'G'); put('G', 'C'); put('T', 'A');
	}};
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(System.in);
		String seq = in.nextLine();
		System.out.println(new StringBuilder(seq).reverse().chars().mapToObj(x -> (char) x).map(COMPLEMENT::get).map(String::valueOf).collect(Collectors.joining()));
		in.close();
	}
}

File: Projects/rosalind/REVC.kt
package rosalind

import java.io.IOException
import java.util.*
import java.util.stream.Collectors

object REVC {
    private val COMPLEMENT: Map<Char, Char> = mapOf(
        'A' to 'T', 'C' to 'G', 'G' to 'C', 'T' to 'A'
    )

    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        val seq = stdin.nextLine()
        println(seq.reversed().map { COMPLEMENT[it] }.joinToString(""))
        stdin.close()
    }
}